Hardware/Hardware/CardReader/CardReaderService.swift

```diff
@@ -38,7 +38,9 @@ public protocol CardReaderService {
 
     /// Captures a payment after collecting a payment method succeeds.
     /// The returned publisher will behave as a Future, eventually producing a single value and finishing, or failing.
-    func capturePayment(_ parameters: PaymentIntentParameters) -> AnyPublisher<PaymentIntent, Error>
+    func capturePayment(_ parameters: PaymentIntentParameters) ->
+    (future: AnyPublisher<PaymentIntent, Error>,
```

Comment:
nit: I am wondering, could we return Combine's `Future` type there instead of `AnyPublisher`? Since we are already revealing the type of `Publisher` we are returning, it would be even more clear if we specify the type.

Reply:
Yea it'd be nice if we can return a `Future` to ensure the one-time completion behavior. Unfortunately, I don't think we can keep the `Future` type after chaining multiple `Future`s with `flatMap` as in the current implementation 😞 I think we could update each step in the chain to a completion block (or even better with async/await when we can use it) and chain them in a single `Future`, but that might be too big of a change. Curious to hear what you think about this and I can create a separate task to refactor this!

Reply:
Ah I see, thanks for the explanation!

>  I think we could update each step in the chain to a completion block (or even better with async/await when we can use it) and chain them in a single Future, but that might be too big of a change.

It is true that we wouldn't keep the `Future` type after chaining it with the `flatMap`. Maybe we can revisit it when we can use `async/await`, perhaps even using an `AsyncSequence` for tackling it :)